spec/controllers/tag_wranglings_controller_spec.rb

```diff
@@ -4,60 +4,146 @@
   include LoginMacros
   include RedirectExpectationHelper
 
-  before do
-    fake_login
-    controller.current_user.roles << Role.new(name: "tag_wrangler")
+  full_access_roles = %w[superadmin tag_wrangling].freeze
+  read_access_roles = %w[superadmin policy_and_abuse tag_wrangling].freeze
+
+  shared_examples "an action only authorized admins can access" do |authorized_roles:|
+    before do
+      fake_login_admin(admin)
+      subject
```

Comment:
I'd prefer the subject call to be in the `it` blocks, since that is what is being tested, but it doesn't really matter.